src/components/valences-test/ValencesTest.tsx

```diff
@@ -16,202 +15,190 @@ import TestResults from "../test-results/TestResults";
-          <div className="valences-test__result">
-            <Card className="valences-test__result-card">
-              <TestResults
-                gaTestName="Valences Test"
-                wrongAnswers={wrong.length}
-                rightAnswers={right.length}
-                onRepeat={this.repeatTest}
-                onRepeatWrongAnswers={this.repeatWrongAnswers}
-              />
-            </Card>
-          </div>
-        )}
-      </div>
-    );
+  function createQuestion(element: IElement): ValencesTestQuestionCard {
+    return {
+      answers: createQuestionAnswers(element),
+      data: element,
+      question: element.symbol,
+      questionClass: `valences-test__question element ${element.group}`,
+    };
   }
 
-  private onNavbarBackButtonClick() {
-    const { history } = this.props;
+  function createTestQuestions() {
+    const questions = settings!
+      .elements!.filter((element) => element.enabled)
+      .map((element) => ElementManager.getElement(element.atomic))
+      .map((element) => createQuestion(element!));
 
-    history.push(TEST_SELECTION);
+    return shuffle(questions);
   }
 
-  private onQuestionAnswer(
-    question: IValencesTestQuestionCard,
+  const [quiz, setQuiz] = React.useState<ValencesTestState>(() => ({
+    questions: createTestQuestions(),
+    right: [],
+    wrong: [],
+  }));
+
+  const { questions, wrong, right } = quiz;
+
+  const hasQuestions = !!questions.length;
+
+  function onQuestionAnswer(
+    question: ValencesTestQuestionCard,
     answer: Answer
   ) {
-    const elementSetting = this.settings.elements!.find(
-      (element) => element.atomic === question.data.atomic
+    const elementSetting = settings.elements!.find(
+      (element: ITestElementSettings) => element.atomic === question.data.atomic
     );
+    if (!elementSetting) return;
 
-    const alreadyAnswered = this.isAlreadyAnswered(question);
+    const alreadyAnswered = isAlreadyAnswered(question);
 
     if (!alreadyAnswered) {
-      elementSetting!.stats.times++;
+      elementSetting.stats.times++;
 
       if (answer.right) {
-        elementSetting!.stats.right++;
-        this.addRightAnsweredQuestion(question);
+        elementSetting.stats.right++;
+        addRightAnsweredQuestion(question);
       } else {
-        elementSetting!.stats.wrong++;
-        this.addWrongAnsweredQuestion(question);
+        elementSetting.stats.wrong++;
+        addWrongAnsweredQuestion(question);
       }
     }
 
     if (answer.right) {
-      this.removeQuestion(question);
+      removeQuestion(question);
     }
 
     AppSettings.save();
   }
 
-  private removeQuestion(question: IValencesTestQuestionCard) {
-    const { questions } = this.state;
-
-    this.setState({
-      questions: questions.filter((value) => value !== question),
+  function updateQuiz(newState: Partial<ValencesTestState>) {
+    setQuiz((oldQuiz) => {
+      return { ...oldQuiz, ...newState };
     });
   }
```

Comment:
Sure we can split, made sense to me to keep it in one, since all of them are "related" to the questions on the quiz, but theres no actual benefit in splitting them